qtpy/Qsci.py

```diff
@@ -0,0 +1,28 @@
+# -----------------------------------------------------------------------------
+# Copyright © 2009- The Spyder Development Team
+#
+# Licensed under the terms of the MIT License
+# (see LICENSE.txt for details)
+# -----------------------------------------------------------------------------
+
+"""Provides Qsci classes and functions."""
+
+from . import (
+    PYQT5,
+    PYQT6,
+    PYSIDE2,
+    PYSIDE6,
+    QtBindingsNotFoundError,
+    QtBindingMissingModuleError,
+)
+
+if PYQT5:
+    from PyQt5.Qsci import *
+elif PYQT6:
+    from PyQt6.Qsci import *
```

Comment:
I think we should add here some logic to handle the extra package that provides the QSintilla bindings not being available/not installed. So something like:
```suggestion
from . import (
    PYQT5,
    PYQT6,
    PYSIDE2,
    PYSIDE6,
    QtBindingsNotFoundError,
    QtBindingMissingModuleError,
    QtModuleNotInstalledError
)

if PYQT5:
    try:
        from PyQt5.Qsci import *
    except ModuleNotFoundError as error:
        raise QtModuleNotInstalledError(
            name='Qsci', missing_package='QScintilla'
        ) from error
elif PYQT6:
    try:
        from PyQt6.Qsci import *
    except ModuleNotFoundError as error:
        raise QtModuleNotInstalledError(
            name='Qsci', missing_package='PyQt6-QScintilla'
        ) from error
```